x/erc20/keeper/ibc_callbacks.go

```diff
@@ -101,15 +100,25 @@ func (k Keeper) OnRecvPacket(
 		return ack
 	}
 
-	pair, _ := k.GetTokenPair(ctx, pairID)
-	if !pair.Enabled {
+	pair, found := k.GetTokenPair(ctx, pairID)
+	if !(found && pair.Enabled) {
 		// no-op: continue with the rest of the stack without conversion
 		return ack
 	}
 
+	address := common.HexToAddress(pair.Erc20Address)
```

Comment:
not sure why u need this code
already called in `GetTokenPairID`

Reply:
It's called above by passing the `coin.Denom` which is not the token address, like here I think. Here we explicitly check if the token is stored by the address